core/engine/src/job.rs

```diff
@@ -115,6 +116,87 @@ impl NativeJob {
     }
 }
 
+/// An ECMAScript [Job] that runs after a certain amount of time.
+///
+/// This represents the [`HostEnqueueTimeoutJob`] operation from the specification.
+///
+/// [HostEnqueueTimeoutJob]: https://tc39.es/ecma262/#sec-hostenqueuetimeoutjob
+pub struct TimeoutJob {
+    /// The instant this job should be run, in msec since epoch. This will be compared
+    /// to the host's [`HostHooks::utc_now`] method.
```

Comment:
I don't think we should mention details about `SimpleJobExecutor`, since every executor can handle the timeout amount in a different way.

Reply:
They should definitely not. People building libraries should not be tied to what an executor interpret the value as.